core/mm.c

```diff
@@ -176,14 +177,22 @@ static void map_os_memory(void)
 	list_for_each_entry(region, &mem_list, list) {
 		type = memory_region_type(region);
 		if ((type != MEMORY_REGION_TYPE_NORMAL) &&
-				(type != MEMORY_REGION_TYPE_DMA))
+		    (type != MEMORY_REGION_TYPE_DMA) &&
+		    /*
+		     * Map the memory in VM region, because we may touch the
+		     * kernel and fdt.
+		     */
+		    (type != MEMORY_REGION_TYPE_VM))
 			continue;
 
 		if (type == MEMORY_REGION_TYPE_DMA)
 			flags |= VM_IO;
 		else
 			flags |= VM_NORMAL;
 
+		if (type == MEMORY_REGION_TYPE_RAMDISK)
+			flags |= VM_RO;
```

Comment:
It's weird that this flag will cause `sync error in EL2`.

Reply:
yes, there is a mistake, it should be : flags |= VM_NORMAL | VM_RO.

Reply:
It is `if` instead of `else if`, which means it is already `VM_NORMAL | VM_RO`.

Reply:
It seems that Minos cannot access ramdisk during `ramdisk_init` if `flags |= VM_RO`.

Reply:
After thinking about it, there will be problems with mapping here. The memory of the ramdisk belongs to the kernel, and all kernel memory has been mapped when it is initialized. The logic of this piece needs to be changed。

Reply:
You use the `S1_` prefix macro in `arch_host_tt_description` function. Should the memory used by Minos belong to stage 2, right?

https://developer.arm.com/architectures/learn-the-architecture/aarch64-memory-model/combining-stage-1-and-stage-2-attributes

Reply:
stage2 is for VM. Minos is stage1. ramdisk memory region has already been mapped in arch/aarch64/core/mem_map.S, there should some issue when change the mapping for those memory which already mapped.

Reply:
The location of the ramdisk does not belong to the kernel, it is not mapped. I have already tried it.

Reply:
ok got it, you load the ramdisk into the memory region which do not belongs to kernel.

Reply:
Yes, that is why we need the `ramdisk` region.

Reply:
which address do you load ? create_host_mapping will map the memory as 2M align?

Reply:
I mapped it to 0x4a000000.

Reply:
list_for_each_entry(region, &mem_list, list) {
                type = memory_region_type(region);
+                /*
+                 * Map the memory in VM region, because we may touch the
+                 * kernel and fdt.
+                 */
                if ((type != MEMORY_REGION_TYPE_NORMAL) &&
-                               (type != MEMORY_REGION_TYPE_DMA))
+                   (type != MEMORY_REGION_TYPE_DMA) &&
+                   (type != MEMORY_REGION_TYPE_VM) &&
+                   (type != MEMORY_REGION_TYPE_RAMDISK));
                        continue;

please try it again.